src/populators/ecoverse-populator.ts

```diff
@@ -31,98 +32,124 @@ export class EcoversePopulator extends AbstractPopulator {
-          //   {
-          //     name: 'repo',
-          //     uri: ecoverse.refRepo,
-          //     description: 'The ecoverse repository',
-          //   },
-          // ],
-        },
-      });
-      await this.client.updateReferencesOnEcoverse([
+      await this.client.updateReferencesOnEcoverse(ecoverseData.nameID, [
         {
           name: 'website',
-          uri: ecoverse.refWebsite,
+          uri: ecoverseData.refWebsite,
           description: 'The ecoverse website',
         },
         {
           name: 'logo',
-          uri: ecoverse.refLogo,
+          uri: ecoverseData.refLogo,
           description: 'The ecoverse logo',
         },
         {
           name: 'repo',
-          uri: ecoverse.refRepo,
+          uri: ecoverseData.refRepo,
           description: 'The ecoverse repository',
         },
       ]);
-
-      this.logger.info(`Ecoverse updated: ${ecoverse.name}`);
     } catch (e) {
       if (e.response && e.response.errors) {
         this.logger.error(
-          `Unable to update ecoverse (${ecoverse.name}):${e.response.errors[0].message}`
+          `Unable to create/update ecoverse (${ecoverseData.nameID}):${e.response.errors[0].message}`
         );
       } else {
         this.logger.error(
-          `Unable to update ecoverse (${ecoverse.name}): ${e.message}`
+          `Unable to create/update ecoverse (${ecoverseData.nameID}): ${e.message}`
         );
       }
     } finally {
       this.profiler.profile(ecoverseProfileID);
     }
   }
+
+  async updateEcoverse(ecoverseData: Ecoverse) {
+    await this.client.updateEcoverse({
+      ID: ecoverseData.nameID,
+      displayName: ecoverseData.displayName,
+      hostID: ecoverseData.host,
+      context: {
+        background: ecoverseData.background,
+        impact: ecoverseData.impact,
+        tagline: ecoverseData.tagline,
+        vision: ecoverseData.vision,
+        who: ecoverseData.who,
+        // references: [
+        //   {
+        //     name: 'website',
+        //     uri: ecoverse.refWebsite,
+        //     description: 'The ecoverse website',
+        //   },
+        //   {
+        //     name: 'logo',
+        //     uri: ecoverse.refLogo,
+        //     description: 'The ecoverse logo',
+        //   },
+        //   {
+        //     name: 'repo',
+        //     uri: ecoverse.refRepo,
+        //     description: 'The ecoverse repository',
+        //   },
+        // ],
+      },
+    });
+
+    this.logger.info(`Ecoverse updated: ${ecoverseData.displayName}`);
+  }
+
+  async createEcoverse(ecoverseData: Ecoverse) {
+    await this.client.createEcoverse({
+      nameID: ecoverseData.nameID,
+      displayName: ecoverseData.displayName,
+      hostID: ecoverseData.host,
+      context: {
+        background: ecoverseData.background,
+        impact: ecoverseData.impact,
+        tagline: ecoverseData.tagline,
+        vision: ecoverseData.vision,
+        who: ecoverseData.who,
+        // references: [
```

Comment:
Same as above.

Reply:
These code pieces were already commented out, and did not have bandwidth to look at the rational so just left them there. Suggest to leave for now as higher priorities...